src/Workspaces/Core/Portable/Storage/CloudCache/CloudCachePersistentStorage.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.IO.Pipelines;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.PersistentStorage;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.Storage
+{
+    /// <summary>
+    /// Implementation of Roslyn's <see cref="IPersistentStorage"/> sitting on top of the platform's cloud storage
+    /// system.
+    /// </summary>
+    internal class CloudCachePersistentStorage : AbstractPersistentStorage
```

Comment:
❔ Can we move these up to the features layer? It's unlikely they'll benefit command-line processing tools and the persistent storage dependencies at this layer make things unnecessarily burdensome for consumers.